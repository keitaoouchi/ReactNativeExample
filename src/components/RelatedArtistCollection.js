import React from "react";
import {Animated, Easing, ScrollView, Text, TouchableOpacity, View} from "react-native";
import Color from "../misc/Color";

export default class RelatedArtistCollection extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fadeAnim: new Animated.Value(0),
    };
  }

  render() {
    const title = this.props.title;
    const artists = this.props.artists;
    const onPress = this.props.onPress;
    return (
      <View style={style.root}>
        <Text style={style.title}>{title}</Text>
        <ScrollView
          horizontal={true}
          style={style.scroll}
          showsVerticalScrollIndicator={false}
          alwaysBounceVertical={false}
          bounces={true}
          directionalLockEnabled={true}
          automaticallyAdjustContentInsets={false}
        >
          {artists.map((artist, i) => (
            <TouchableOpacity
              key={i}
              onPress={() => onPress(artist)}
              style={{
                flex: 1,
                alignItems: 'center',
                justifyContent: 'flex-start',
              }}
            >
              <Animated.Image
                source={{uri: artist.imageURL}}
                style={[style.image, {opacity: this.state.fadeAnim}]}
                onLoadEnd={() => {
                  Animated.timing(
                    this.state.fadeAnim,
                    {
                      toValue: 1,
                      easing: Easing.easeInOut,
                    },
                  ).start();
                }}
              />
              <Text
                style={style.artistName}
                numberOfLines={2}
              >{artist.name}</Text>
            </TouchableOpacity>),
          )}
        </ScrollView>
      </View>
    );
  }
}

const style = {
  root: {
    marginBottom: 8,
  },
  scroll: {height: 180},
  title: {
    fontSize: 16,
    color: 'white',
    margin: 8,
  },
  image: {
    width: 120,
    height: 120,
    margin: 8,
  },
  artistName: {
    color: 'white',
    textAlign: 'center',
    fontWeight: 'bold',
    backgroundColor: Color.black,
    opacity: 0.6,
    width: 120,
    height: 36,
    left: 0,
    right: 0,
    fontSize: 14,
  },
};
