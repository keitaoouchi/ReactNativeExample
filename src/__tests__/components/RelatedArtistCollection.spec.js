import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import RelatedArtistCollection from "../../../src/components/RelatedArtistCollection";

describe('RelatedArtistCollection', () => {

  it('renders correctly', () => {
    const tree = renderer.create(
      <RelatedArtistCollection title="test" artists={[]} onPress={() => {
      }}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});