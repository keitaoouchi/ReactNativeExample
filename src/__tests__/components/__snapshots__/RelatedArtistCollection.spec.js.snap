// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedArtistCollection renders correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 8,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 16,
        "margin": 8,
      }
    }
  >
    test
  </Text>
  <RCTScrollView
    alwaysBounceVertical={false}
    automaticallyAdjustContentInsets={false}
    bounces={true}
    directionalLockEnabled={true}
    horizontal={true}
    showsVerticalScrollIndicator={false}
    style={
      Object {
        "height": 180,
      }
    }
  >
    <View />
  </RCTScrollView>
</View>
`;
